base/show.jl

```diff
@@ -1657,17 +1660,15 @@ function show_vector(io::IO, v, opn, cls)
         io = IOContext(io, :compact => compact)
     end
     print(io, prefix)
-    if limited && length(v) > 20
-        inds = _indices1(v)
+    if limited && _length(v) > 20
```

Comment:
http://docs.julialang.org/en/latest/devdocs/offset-arrays/#do-not-implement-size-or-length
http://docs.julialang.org/en/latest/devdocs/offset-arrays/#generalizing-existing-code

(This is a temporary state, we will go back to supporting `length` someday.)